### PR TITLE
:sparkles: Add Script Injection to Dangerous-Workflow

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

* **Please check if the PR fulfills these requirements**
- [ x] Tests for the changes have been added (for bug fixes / features)
- [ x] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature: Adds script injection to Dangerous-Workflow check
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections

* **What is the current behavior?** (You can also link to an open issue here)
part of https://github.com/ossf/scorecard/issues/426

* **What is the new behavior (if this is a feature change)?**
Adds a new check. 
**feedback** would like feedback on the scoring for this check. Right now if any fails, you get a 0

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by asraa** [comment]
cc @haydentherapper just fyi!

**Review by azeemshaikh38 — APPROVED** [review]
LGTM. Would be good to have some more unit tests for untrusted workflows.

**Comment by asraa** [comment]
added more tests!

File: checks/dangerous_workflow.go [hunk 1]
```diff
@@ -16,17 +16,47 @@ package checks
...
 	"github.com/ossf/scorecard/v3/checker"
 	"github.com/ossf/scorecard/v3/checks/fileparser"
+	sce "github.com/ossf/scorecard/v3/errors"
 )
 
 // CheckDangerousWorkflow is the exported name for Dangerous-Workflow check.
 const CheckDangerousWorkflow = "Dangerous-Workflow"
 
+func containsUntrustedContextPattern(variable string) bool {
+	// regexp for untrusted code patterns github.event.
```
Comment (laurentsimon) [inline checks/dangerous_workflow.go:33]: nit: start with capital letter al comments

File: checks/dangerous_workflow.go [hunk 2]
```diff
@@ -16,17 +16,47 @@ package checks
...
+			`commits.*\.author\.email|` +
+			`commits.*\.author\.name|` +
+			`pull_request\.head\.ref|` +
+			`pull_request\.head\.label|` +
+			`pull_request\.head\.repo\.default_branch).*`)
+
+	if strings.Contains(variable, "github.head_ref") {
+		return true
+	}
+	fmt.Printf("%s %t\n", variable, untrustedContextPattern.MatchString(variable))
```
Comment (laurentsimon) [inline checks/dangerous_workflow.go:56]: don't forget to remove.

File: checks/dangerous_workflow.go [hunk 3]
```diff
@@ -16,17 +16,47 @@ package checks
...
 
 	"github.com/ossf/scorecard/v3/checker"
 	"github.com/ossf/scorecard/v3/checks/fileparser"
+	sce "github.com/ossf/scorecard/v3/errors"
 )
 
 // CheckDangerousWorkflow is the exported name for Dangerous-Workflow check.
 const CheckDangerousWorkflow = "Dangerous-Workflow"
 
+func containsUntrustedContextPattern(variable string) bool {
```
Comment (laurentsimon) [inline checks/dangerous_workflow.go:32]: can you add as comment the link where you took those from? Either the security lab blog pst of the GitHub doc would do, I think.

**Review by laurentsimon — APPROVED** [review]
This is exciting!

**Comment by laurentsimon** [comment]
Thanks @asraa !

**Pull request merged**